erpnext/payroll/doctype/gratuity/gratuity.py

```diff
@@ -0,0 +1,190 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020, Frappe Technologies Pvt. Ltd. and contributors
+# For license information, please see license.txt
+
+from __future__ import unicode_literals
+import frappe
+from frappe import _, bold
+from frappe.model.document import Document
+from frappe.utils import flt, get_datetime, get_link_to_form
+from math import floor
+
+from frappe.utils import get_datetime
+class Gratuity(Document):
+	def validate(self):
+		data = calculate_work_experience_and_amount(self.employee, self.gratuity_rule)
+		self.current_work_experience = data["current_work_experience"]
+		self.amount = data["amount"]
+
+	def before_submit(self):
+		self.status = "Unpaid"
+		if self.pay_via_salary_slip:
+			self.status = "Paid"
+
+	def on_submit(self):
+		if self.pay_via_salary_slip:
+			additional_salary = frappe.new_doc('Additional Salary')
+			additional_salary.employee = self.employee
+			additional_salary.salary_component = self.salary_component
+			additional_salary.overwrite_salary_structure_amount = 0
+			additional_salary.amount = self.amount
+			additional_salary.payroll_date = self.payroll_date
+			additional_salary.company = self.company
+			additional_salary.ref_doctype = self.doctype
+			additional_salary.ref_docname = self.name
+			additional_salary.submit()
```

Comment:
we should avoid writing logic directly inside `on_submit` function. Move to a separate function.

Reply:
done